src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php

```diff
@@ -0,0 +1,197 @@
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Category\QueryHandler;
+
+use Category;
+use ImageManager;
+use ImageType;
+use PrestaShop\PrestaShop\Core\Domain\Category\EditableCategory;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Category\Query\GetCategoryForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Category\QueryHandler\GetCategoryForEditingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
+use PrestaShop\PrestaShop\Core\Image\Parser\ImageTagSourceParserInterface;
+
+/**
+ * Class GetCategoryForEditingHandler.
+ */
+final class GetCategoryForEditingHandler implements GetCategoryForEditingHandlerInterface
+{
+    /**
+     * @var ImageTagSourceParserInterface
+     */
+    private $imageTagSourceParser;
+
+    /**
+     * @param ImageTagSourceParserInterface $imageTagSourceParser
+     */
+    public function __construct(ImageTagSourceParserInterface $imageTagSourceParser)
+    {
+        $this->imageTagSourceParser = $imageTagSourceParser;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CategoryNotFoundException
+     */
+    public function handle(GetCategoryForEditing $query)
+    {
+        $category = new Category($query->getCategoryId()->getValue());
+
+        if (!$category->id) {
+            throw new CategoryNotFoundException(
+                $query->getCategoryId(),
+                sprintf('Category with id "%s" was not found', $query->getCategoryId()->getValue())
+            );
+        }
+
+        $editableCategory = new EditableCategory(
+            $category->name,
+            (bool) $category->active,
+            $category->description,
+            $category->id_parent,
+            $category->meta_title,
+            $category->meta_description,
+            $category->meta_keywords,
+            $category->link_rewrite,
+            $category->getGroups(),
+            $category->getAssociatedShops(),
+            (bool) $category->is_root_category,
+            $this->getCoverImage($query->getCategoryId()),
+            $this->getThumbnailImage($query->getCategoryId()),
+            $this->getMenuThumbnailImages($query->getCategoryId())
+        );
+
+        return $editableCategory;
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getCoverImage(CategoryId $categoryId)
+    {
+        $imageType = 'jpg';
+        $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.' . $imageType;
+
+        $imageTag = ImageManager::thumbnail(
+            $image,
+            'category' . '_' . $categoryId->getValue() . '.' . $imageType,
+            350,
+            $imageType,
+            true,
+            true
+        );
+
+        $imageSize = file_exists($image) ? filesize($image) / 1000 : '';
+
+        if (empty($imageTag) || empty($imageSize)) {
+            return null;
```

Comment:
The doc says it must return an array. Throw exception, return an empty array or change doc block please :)